src/codewhisperer/util/supplementalContext/crossFileContextUtil.ts

```diff
@@ -158,29 +181,19 @@ function splitFileToChunks(filePath: string, chunkSize: number): Chunk[] {
  * This function will return relevant cross files for the given editor file
  * by referencing open files, imported files and same package files.
  */
-async function getRelevantCrossFiles(editor: vscode.TextEditor, dependencyGraph: DependencyGraph): Promise<string[]> {
-    return getOpenFilesInWindow().filter(file => {
-        return isRelevant(editor.document.fileName, file, editor.document.languageId)
+async function getCrossFileCandidates(editor: vscode.TextEditor): Promise<string[]> {
+    const targetFileName = editor.document.fileName
+    const language = editor.document.languageId
+
+    return getOpenFilesInWindow(async candidateFile => {
+        return (
+            targetFileName !== candidateFile &&
+            path.extname(targetFileName) === path.extname(candidateFile) &&
+            !(await isTestFile(candidateFile, { languageId: language }))
+        )
     })
 }
 
-function getOpenFilesInWindow(): string[] {
-    const filesOpenedInEditor: string[] = []
-
-    try {
-        const tabArrays = vscode.window.tabGroups.all
-        tabArrays.forEach(tabArray => {
-            tabArray.tabs.forEach(tab => {
-                filesOpenedInEditor.push((tab.input as any).uri.path)
-            })
-        })
-    } catch (e) {
-        // Older versions of VSC do not have the tab API
-    }
-
-    return filesOpenedInEditor
-}
-
```

Comment:
Moved to supplementalContextUtil.ts as it will be used by both crossfile and utg features 

https://github.com/aws/aws-toolkit-vscode/pull/3668/files#diff-752c5d36a246334f6da7cd1a91d445bd971aa71573a337a51c293e5cc1ac53a4R84